packages/pyright-internal/src/analyzer/operations.ts

```diff
@@ -215,34 +220,46 @@ export function validateBinaryOperation(
             return { type: NeverType.createNever() };
         }
 
+        const validateArithmetic = (context: InferenceContext | undefined, addendum: DiagnosticAddendum | undefined) =>
+            validateArithmeticOperation(
+                evaluator,
+                operator,
+                leftTypeResult,
+                rightTypeResult,
+                errorNode,
+                context,
+                addendum,
+                options
+            );
+
         // Handle certain operations on certain homogenous literal types
         // using special-case math. For example, Literal[1, 2] + Literal[3, 4]
         // should result in Literal[4, 5, 6].
         if (options.isLiteralMathAllowed) {
             type = calcLiteralForBinaryOp(operator, leftType, rightType);
+            const result = validateArithmetic(undefined, undefined);
+            if (type) {
+                overloadsUsedForCall.push(...(result?.overloadsUsedForCall ?? []));
+            }
         }
 
         if (!type) {
-            const result = validateArithmeticOperation(
-                evaluator,
-                operator,
-                leftTypeResult,
-                rightTypeResult,
-                errorNode,
-                inferenceContext,
-                diag,
-                options
-            );
+            const result = validateArithmetic(inferenceContext, diag);
 
             if (result.magicMethodDeprecationInfo) {
                 deprecatedInfo = result.magicMethodDeprecationInfo;
             }
+            overloadsUsedForCall.push(...(result?.overloadsUsedForCall ?? []));
```

Comment:
i dont think `result` can be `undefined` here
```suggestion
            overloadsUsedForCall.push(...(result.overloadsUsedForCall ?? []));
```

Reply:
You’re right, this is fixed now.